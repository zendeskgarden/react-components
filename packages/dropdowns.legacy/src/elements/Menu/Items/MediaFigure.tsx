/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { StyledMediaFigure } from '../../../styled';
import useMenuContext from '../../../utils/useMenuContext';

/**
 * @extends HTMLAttributes<Element>
 */
export const MediaFigure = (props: HTMLAttributes<Element>) => {
  const { isCompact } = useMenuContext();

  return <StyledMediaFigure isCompact={isCompact} {...props} />;
};
