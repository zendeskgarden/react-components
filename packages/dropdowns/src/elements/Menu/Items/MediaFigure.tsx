/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { StyledMediaFigure } from '../../../styled';
import useMenuContext from '../../../utils/useMenuContext';

/**
 * Accepts all `<div>` props
 */
export const MediaFigure: React.FC<HTMLAttributes<any>> = props => {
  const { isCompact } = useMenuContext();

  return <StyledMediaFigure isCompact={isCompact} {...props} />;
};
