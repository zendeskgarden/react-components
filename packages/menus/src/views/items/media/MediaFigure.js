import React, { Children } from 'react';
import MenuStyles from '@zendesk/garden-css-menus';

import { version } from '../../../../package.json';
const COMPONENT_ID = 'menus.media_figure';

const MediaFigure = ({ children }) => {
  return React.cloneElement(Children.only(children), {
    'data-garden-id': COMPONENT_ID,
    'data-garden-version': version,
    className: MenuStyles['c-menu__item--media__figure']
  });
};

/** @component */
export default MediaFigure;
