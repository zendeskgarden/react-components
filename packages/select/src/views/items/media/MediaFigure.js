import { MediaFigure } from '@zendesk/garden-react-menus';

/** @component */
export default MediaFigure;
