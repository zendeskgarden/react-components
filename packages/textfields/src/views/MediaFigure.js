/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import styled from 'styled-components';
import { retrieveTheme } from '@zendeskgarden/react-theming';
import TextStyles from '@zendeskgarden/css-forms/dist/text.css';

const COMPONENT_ID = 'textfields.media_figure';

/**
 * Accepts all `<div>` props
 */
const MediaFigure = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  className: TextStyles['c-txt__input--media__figure']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default MediaFigure;
