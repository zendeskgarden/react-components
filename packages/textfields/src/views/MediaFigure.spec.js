import React from 'react';
import { shallow } from 'enzyme';

import MediaFigure from './MediaFigure';

describe('MediaFigure', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<MediaFigure />);

    expect(wrapper).toMatchSnapshot();
  });
});
