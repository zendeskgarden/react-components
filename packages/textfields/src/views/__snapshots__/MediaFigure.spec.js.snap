// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaFigure renders default styling 1`] = `
<div
  className="c-txt__input--media__figure "
  data-garden-id="textfields.media_figure"
  data-garden-version="version"
/>
`;
